src/client/datascience/notebook/cellOutputs.ts

```diff
@@ -0,0 +1,379 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { nbformat } from '@jupyterlab/coreutils';
+import cloneDeep from 'lodash/cloneDeep';
+import { NotebookCellOutput, NotebookCellOutputItem } from 'vscode';
+import { concatMultilineString, splitMultilineString } from '../../../datascience-ui/common';
+import { traceWarning } from '../../common/logger';
+import { sendTelemetryEvent } from '../../telemetry';
+import { Telemetry } from '../constants';
+import { CellOutputMimeTypes } from './types';
+
+/**
+ * Metadata we store in VS Code cell output items.
+ * This contains the original metadata from the Jupyter Outputs (nbformat.IOutput).
+ */
+export interface CellOutputMetadata {
```

Comment:
Created concrete classes for 
* Outputs that display the following jupyter outputs
	* nbformat.IError
	* nbformat.IDisplay
	* nbformat.IUpdateDisplay
	* nbformat.IExecuteReulst
This way its easier to understand the code, else we have functions that cast objects into different types.

* Ensure we support other kernels that add outputs/cells to teh notebook.
E.g. if .NET adds output, we'll need to save them properly in ipynb file. 
Now we save unrecognized outputs as `nbformat.IUnrecognizedOutput`

* I understand we do'nt want classes in the same file, but i kept themn here as they are small & there's a lot of code that gets re-used.
Happy to split into separate files.